clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4806,7 +4275,7 @@ public ListTransactionsResult listTransactions(ListTransactionsOptions options)
             @Override
             void handleResponse(AbstractResponse response) {
                 long currentTimeMs = time.milliseconds();
-                driver.onResponse(currentTimeMs, spec, response);
+                driver.onResponse(currentTimeMs, spec, response, nodeProvider.provide());
```

Comment:
All of this is specifically because `ConsumerGroupDescription` needs the coordinator as a `Node` when describing consumer groups.

At the moment, handlers only get the brokerId and that's not enough to build a `Node` object. So instead in this PR, I replaced the `brokerId` argument by the full `Node` object. The `brokerId` can be retrieved from `spec.scope` but for the `Node`, we need to get that from the `Call` itself. 

Good catch, `nodeProvider.provide()` could return a different `Node`! As far as I can tell, we can use `curNode()` here, that should have been set by `maybeDrainPendingCall()`.

Reply:
I see. That seems reasonable to me now.